### PR TITLE
logseq: add darwin platform

### PR DESCRIPTION
**Pull request by gdw2**
###### Description of changes
Add darwin platform for logseq

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gdw2** [comment]
This is my first ever nix pkg contribution.  I added darwin by looking at what was done [here](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/applications/networking/instant-messengers/slack/default.nix#L81) and am very open to feedback.  

I'd like to update the pkg version to 0.8.6, but am not sure how to generate the sha for linux.  The only technique that worked for me for getting the `dmg` shaw was to try to build the derivation and then apply the feedback about the sha in the error message.  Because I don't have a linux machine to build on, I couldn't easily apply the same technique for the linux sha.

File: pkgs/applications/misc/logseq/default.nix [hunk 1]
```diff
@@ -1,59 +1,85 @@
...
-    url = "https://github.com/logseq/logseq/releases/download/${version}/logseq-linux-x64-${version}.AppImage";
-    sha256 = "sha256-1nvkjucMRAwpqg2LI+1UrICMLzSd6t0yGnYdCUNQslU=";
-    name = "${pname}-${version}.AppImage";
+  meta = with lib; {
+    description =
+      "A local-first, non-linear, outliner notebook for organizing and sharing your personal knowledge base";
+    homepage = "https://github.com/logseq/logseq";
+    license = licenses.agpl3Plus;
+    maintainers = with maintainers; [ weihua ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" ];
```
Comment (risicle) [inline pkgs/applications/misc/logseq/default.nix:17]:
```suggestion
    platforms = [ "x86_64-linux" "x86_64-darwin" ];
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
```

**Comment by kirillrdy** [comment]
hi, so I just released ( was reviewing one of PRs) that there are 3 PRs releated to adding logseq darwin 

https://github.com/NixOS/nixpkgs/pull/268249
https://github.com/NixOS/nixpkgs/pull/248090
https://github.com/NixOS/nixpkgs/pull/191091

can I ask you to collaborate on a new PR ? feel free to reach out for help

looking forward on getting this merged !

**Comment by klchen0112** [comment]
> hi, so I just released ( was reviewing one of PRs) that there are 3 PRs releated to adding logseq darwin
> 
> #268249 #248090 #191091
> 
> can I ask you to collaborate on a new PR ? feel free to reach out for help
> 
> looking forward on getting this merged !

I'm willing to do this. But threre are some questions. 

1. It's necessary to use makeWrapper in darwin? [#268249](https://github.com/NixOS/nixpkgs/pull/268249)
2. For linux, I think this is enough.  [#248090](https://github.com/NixOS/nixpkgs/pull/248090) add `curl expat openssl zlib` in buildinputs
```nix
nativeBuildInputs = [ makeWrapper autoPatchelfHook ];
buildInputs = [ stdenv.cc.cc];
```
3. `runHook preInstall` and `runHook postInstall` should be added for darwin build?

**Comment by yzx9** [comment]
Closing as #331824 has been merged, which adds drawing support.

**Pull request closed without merging**